NewHaven/test/Part-3_4-Test.cpp

```diff
@@ -0,0 +1,55 @@
+//
+// Created by Muher on 2020-03-25.
+//
+#include <iostream>
+#include "Player.h"
+
+int main() {
+    auto *bDeck = new BuildingDeck();
+    auto *hDeck = new HarvestDeck();
+
+    // Initiating Player and their hand to replicate the game for just this part
+    // In practice, this part comes to play later in a turn, not at the beginning.
+
+    // Assume a 2-player game
+    auto *player1 = new Player("Player1");
+    auto *player2 = new Player("Player2");
+
+    // Start of the game, each player should have 6 buildings and 2 Harvest Tiles
+    for (int i = 0; i < 6; i ++){
+        (*player1).drawBuilding(bDeck->draw());
+        (*player2).drawBuilding(bDeck->draw());
+    }
+    for (int i = 0; i < 2; i++){
+        (*player1).drawHarvestTile(hDeck->draw());
+        (*player2).drawHarvestTile(hDeck->draw());
+    }
+
+    // Temporary trail to do this part, would use the actual trail in practice
+    auto trail = new ResourceTrails;
+
+    // Calculate Player1 's resource, this is the LEFTOVER resources other players can now use
```

Comment:
The logic here isn't wrong, but really you should have a vector of players. Start at one index, and loop until the resource tracker reaches 0 in all components or until you reach the turn player again. 

Otherwise, its good if you can into account the changes that would be required from the previous comment; In theory what you have right now looks like it should work, but we should avoid recreatng functions when some exist that can already serve that purpose.